src/peripheral/lift_controller_node.cpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file  lift_controller_node.cpp
+ * @brief This file contains the implementation of a simple action and service
+ *        server for lift controller node. 
+ *
+ * @date  09-01-2024
+ *
+ * @copyright Copyright (c) 2024 Weston Robot Pte. Ltd.
+ */
+#include "wrp_ros/peripheral/lift_controller_node.hpp"
+
+namespace westonrobot {
+LiftControllerNode::LiftControllerNode()
+    : liftController_(),
+      Server_(
+          nh_, "lift_controller_action",
+          [this](const wrp_ros::LiftControlGoalConstPtr& goal) {
+            this->LiftControllerCallback(goal);
+          },
+          false) {
+  if (!ReadParameters()) {
+    ROS_ERROR("Could not load parameters");
+    ros::shutdown();
+  }
+
+  liftController_.Connect(device_path_, baud_rate_);
+  if (!liftController_.IsOkay()) {
+    std::cerr << "Failed to connect to lift controller" << std::endl;
+    ros::shutdown();
+  }
+
+  srvServer_ = nh_.advertiseService("/lift_controller/service",
+                                    &LiftControllerNode::QueryCallback, this);
+  Server_.start(); 
+}
```

Comment:
Maybe move the Server_ below the connect. There is a case here that a action goal arrives before you connect to the lift successfully.

```suggestion
      {
  if (!ReadParameters()) {
    ROS_ERROR("Could not load parameters");
    ros::shutdown();
  }

  liftController_.Connect(device_path_, baud_rate_);
  if (!liftController_.IsOkay()) {
    std::cerr << "Failed to connect to lift controller" << std::endl;
    ros::shutdown();
  }

  srvServer_ = nh_.advertiseService("/lift_controller/service",
                                    &LiftControllerNode::QueryCallback, this);
                                   
  Server_ = actionlib::SimpleActionServer<wrp_ros::LiftControlAction>(
      nh_, "lift_controller_action",
      [this](const wrp_ros::LiftControlGoalConstPtr& goal) {
        this->LiftControllerCallback(goal);
      },
      false);
  Server_.start(); 
}
```